core/storage/question/gae_models.py

```diff
@@ -247,22 +247,25 @@ def get_question_skill_links_and_skill_descriptions(
         return question_skill_link_models, skill_descriptions, next_cursor_str
 
     @classmethod
-    def get_question_skill_links_with_constant_number_per_skill(
-            cls, question_count, skill_ids):
+    def get_question_skill_links_equidistributed_by_skill(
+            cls, total_question_count, skill_ids):
         """Fetches the list of constant number of QuestionSkillLinkModels
         linked to the skills.
 
         Args:
-            question_count: int. The number of questions to be returned.
+            total_question_count: int. The number of questions expected.
             skill_ids: list(str). The ids of skills for which the linked
                 question ids are to be retrieved.
 
         Returns:
-            list(QuestionSkillLinkModel). Constant number of
-                QuestionSkillLinkModels corresponding to given skill_ids.
+            list(QuestionSkillLinkModel). A list of QuestionSkillLinkModels
+                corresponding to given skill_ids, with
+                total_question_count/len(skill_ids) number of questions for
```

Comment:
Description feels incomplete, it's missing stuff about divisibility for example. See comments above about very clearly specifying the return value. This is the contract for tests so you can't leave it imprecise.

Reply:
Done.